src/Illuminate/Support/Fluent.php

```diff
@@ -36,7 +37,11 @@ class Fluent implements Arrayable, ArrayAccess, Jsonable, JsonSerializable
      */
     public function __construct($attributes = [])
     {
-        $this->fill($attributes);
+        if (! $attributes instanceof Traversable) {
+            $attributes = is_object($attributes) ? get_object_vars($attributes) : $attributes;
+        }
+
+        $this->attributes = is_array($attributes) ? $attributes : iterator_to_array($attributes);
```

Comment:
Using `fill()` makes it inline with Eloquent Model: https://github.com/laravel/framework/blob/22f61b33090b57ab154e840752b0d2dbd0a7a3d7/src/Illuminate/Database/Eloquent/Model.php#L278